src/pool/srv_pool.c

```diff
@@ -3190,3 +3211,65 @@ ds_pool_replicas_update_handler(crt_rpc_t *rpc)
 	out->pmo_rc = rc;
 	crt_reply_send(rpc);
 }
+
+static inline struct pl_obj_shard*
+ds_pool_get_shard(void *data, int idx)
+{
+	struct pl_obj_layout	*layout = data;
+
+	return &layout->ol_shards[idx];
+}
+
+int
+ds_pool_check_leader(uuid_t pool_uuid, daos_unit_oid_t *oid,
+		     uint32_t version, struct pl_obj_layout **plo)
+{
+	struct ds_pool		*pool;
+	struct pl_obj_layout	*layout = NULL;
+	struct pool_target	*target;
+	struct daos_obj_md	 md = { 0 };
+	d_rank_t		 leader;
+	d_rank_t		 myrank;
+	int			 rc;
+
+	pool = ds_pool_lookup(pool_uuid);
+	if (pool == NULL)
+		return -DER_INVAL;
+
+	md.omd_id = oid->id_pub;
+	md.omd_ver = version;
+	ABT_rwlock_rdlock(pool->sp_lock);
+	D_ASSERT(pool->sp_pl_map != NULL);
+
+	rc = pl_obj_place(pool->sp_pl_map, &md, NULL, &layout);
+	ABT_rwlock_unlock(pool->sp_lock);
+	if (rc != 0)
+		goto out;
+
+	rc = pl_select_leader(oid, layout->ol_nr, true, ds_pool_get_shard,
+			      layout, &leader);
+	if (rc != 0)
```

Comment:
minor, some log in error case might be helpful to track issue in failure case. same for other places.

Reply:
OK